mhs/common/mhs_common/messages/ebxml_request_envelope.py

```diff
@@ -54,19 +113,39 @@ def from_string(cls, headers: Dict[str, str], message: str) -> EbxmlRequestEnvel
         :return: An instance of an EbxmlAckEnvelope constructed from the message.
         """
         msg = EbxmlRequestEnvelope._parse_mime_message(headers, message)
-        ebxml_part, payload_part = EbxmlRequestEnvelope._extract_message_parts(msg)
-        xml_tree = ElementTree.fromstring(ebxml_part)
+        ebxml_part, payload_part, attachments = EbxmlRequestEnvelope._extract_message_parts(msg)
+        xml_tree: Element = ElementTree.fromstring(ebxml_part)
         extracted_values = super().parse_message(xml_tree)
 
         cls._extract_more_values_from_xml_tree(xml_tree, extracted_values)
 
-        logger.info('0001', 'Extracted {extracted_values} from message', {'extracted_values': extracted_values})
+        cls._add_descriptions_to_attachments(xml_tree, attachments)
+        extracted_values[ATTACHMENTS] = attachments
+
+        logger.info('0002', 'Extracted {extracted_values} from message', {'extracted_values': extracted_values})
 
         if payload_part:
             extracted_values[MESSAGE] = payload_part
 
         return EbxmlRequestEnvelope(extracted_values)
 
+    @classmethod
+    def _add_descriptions_to_attachments(cls, xml_tree, attachments):
+        for attachment in attachments:
+            description_element = xml_tree.find(
+                f".//{ebxml_envelope.EBXML_NAMESPACE}:Reference["
+                f"@{ebxml_envelope.XLINK_NAMESPACE}:href='cid:{attachment[ATTACHMENT_CONTENT_ID]}']"
+                f"/{ebxml_envelope.EBXML_NAMESPACE}:Description",
+                ebxml_envelope.NAMESPACES)
+            if description_element is None:
+                logger.info('0003', "{Attachment} with {ContentType} found with no description. "
+                                    "Setting description to ''",
```

Comment:
Around the description, I've made it mandatory when serialising, and when deserialising, as it's optional in the EIS (see links to EIS on the [JIRA story](https://gpitbjss.atlassian.net/browse/RT-107)), I've put this code here to handle it being optional by setting it as `''`.

@colin-bateson do you have any thoughts about this, do you think I should do things differently?